src/app/table-module/services/table.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Product } from './model';
+import { tap, map } from 'rxjs/operators';
+
+const CREATE_ACTION = 'create';
+const UPDATE_ACTION = 'update';
+const REMOVE_ACTION = 'destroy';
+
+
+@Injectable({
+  providedIn: "root",
+})
+export class TableService extends BehaviorSubject<PersonInterface[]> {
+   constructor(private http: HttpClient) {
+        super([]);
+    }
+
+    private data: Product[] = [];
+
+    public read(): void {
+        if (this.data.length) {
+            return super.next(this.data);
+        }
+
+        this.fetch()
+            .pipe(
+                tap(data => {
+                    this.data = data;
+                })
+            )
+            .subscribe(data => {
+                super.next(data);
+            });
+    }
+
+    public save(data: Product[], isNew?: boolean): void {
+        const action = isNew ? CREATE_ACTION : UPDATE_ACTION;
+
+        this.reset();
+
+        this.fetch(action, data)
+            .subscribe(() => this.read(), () => this.read());
+    }
```

Comment:
purpose of two `this.read()` ?

Reply:
im not doin any thin this is the fuction that in the kendo angular file

Reply:
remove it